core/queryalgebra/model/src/main/java/org/eclipse/rdf4j/query/algebra/Slice.java

```diff
@@ -29,10 +29,11 @@
 	 *--------------*/
 
 	public Slice() {
+		this(null);
 	}
 
 	public Slice(TupleExpr arg) {
-		this(arg, 0, -1);
+		this(arg, -1, -1);
```

Comment:
Changes to fix default (unset) value to -1. This is the same as that used by TupleExprBuilder and ensures equals() works properly no matter which constructor is used.